samples/camera/camerax/src/main/java/com/example/platform/camerax/basic/CameraXBasic.kt

```diff
@@ -0,0 +1,308 @@
+                    modifier = modifier,
+                    lifecycleOwner = LocalLifecycleOwner.current,
+                    onTakePhotoClick = {
+                        viewModel.takePhoto(
+                            context = context,
+                            callbackExecutor = imageCaptureCallbackExecutor,
+                            onImageCaptured = { uri -> onShowCapturedImageChange(uri) },
+                            onError = { /* Error logging/toast handled within takePhoto */ },
+                        )
+                    },
+                )
+            }
+        }
+
+        is PermissionStatus.Denied -> CameraPermissionDeniedView(
+            cameraPermissionState.status,
+            cameraPermissionState,
+        )
+    }
+}
+
+/**
+ * Composable function that displays a message when camera permission is denied.
+ *
+ * It provides an option to request the permission again.
+ *
+ * @param status The current [PermissionStatus] of the camera permission.
+ * @param cameraPermissionState The [PermissionState] for the camera permission, used to request permission again.
+ */
+@Composable
+@OptIn(ExperimentalPermissionsApi::class)
+private fun CameraPermissionDeniedView(
+    status: PermissionStatus,
+    cameraPermissionState: PermissionState,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+    ) {
+        val textToShow = if (status.shouldShowRationale) {
+            "The camera is needed to take pictures. Please grant the permission."
+        } else {
+            "Camera permission is required to use this feature."
+        }
+        Text(text = textToShow)
+        Spacer(modifier = Modifier.height(8.dp))
+        Button(onClick = { cameraPermissionState.launchPermissionRequest() }) {
+            Text("Request Permission")
+        }
+    }
+}
+
+/**
+ * Composable function that displays the camera preview.
+ *
+ * @param viewModel The [CameraXBasicViewModel] for accessing the camera preview surface request.
+ * @param modifier Modifier to be applied to the layout.
+ * @param onTakePhotoClick Callback function to be invoked when the "Take Photo" button is clicked.
+ * @param lifecycleOwner The lifecycle owner to bind the camera to.
+ */
+@Composable
+private fun CameraPreviewContent(
+    viewModel: CameraXBasicViewModel,
+    modifier: Modifier = Modifier,
+    onTakePhotoClick: () -> Unit,
+    lifecycleOwner: LifecycleOwner,
+) {
+    val surfaceRequest by viewModel.surfaceRequest.collectAsStateWithLifecycle()
+    val context = LocalContext.current
+    LaunchedEffect(lifecycleOwner) {
+        viewModel.bindToCamera(context.applicationContext, lifecycleOwner)
+    }
+
+    // State for managing the autofocus indicator
+    var autofocusRequest by remember { mutableStateOf(randomUUID() to Offset.Unspecified) }
```

Comment:
Why is the pair needed here? It seems like this could just be

`var autofocusLocation = remember { mutableStateOf { Offset.Unspecified } {` ?

Am I missing something?

Reply:
This is needed to remember the state of the autofocus point to adjust when the tap-to-focus actually happened

We do the same here: https://gist.github.com/JolandaVerhoef/74d4696b804736c698450bd34b5c9ff8#file-2_1_tap_to_focus-kt-L130